### ISSUE TITLE
Marking multiple episodes watched erroneously scrobbles each episode twice

### ISSUE THREAD

**Issue by blackwind** [MediaBrowser/plugin.video.emby#344]
Using the Emby Trakt plugin (NOT the Kodi one), marking multiple episodes watched through Emby for Kodi at the same time (whether by marking an entire show or just a season) results in each episode being scrobbled twice. Watching an episode or marking episodes one-by-one doesn't exhibit this issue, nor does marking multiple episodes through Emby itself.

In addition, marking multiple episodes watched also scrobbles them in random order, as opposed to sequentially when done through Emby itself.

**Comment by LukePulverenti** [comment]
Hi there, can you please attach the Kodi log? thanks.

**Comment by blackwind** [comment]
I took a log before marking a series watched and after. Here's the diff:

```
2020-05-22 00:14:59.811 T:140701488047872   ERROR: SQL: [Textures13.db] SQLite error SQLITE_CORRUPT (database disk image is malformed)
                                             Query: INSERT INTO texture (id, url, cachedurl, imagehash, lasthashcheck) VALUES(NULL, 'https://EMBY-SERVER/emby/Items/56871/Images/Backdrop/0?EnableImageEnhancers=True&Tag=226f1b4ec48ff036e0082a85d4dd1f9e|verifypeer=false', 'a/ac453749.jpg', 'd1587878875s241283', '') (database disk image is malformed)
2020-05-22 00:14:59.812 T:140701488047872   ERROR: AddCachedTexture failed on url 'https://EMBY-SERVER/emby/Items/56871/Images/Backdrop/0?EnableImageEnhancers=True&Tag=226f1b4ec48ff036e0082a85d4dd1f9e|verifypeer=false'
2020-05-22 00:15:23.680 T:140701446084352  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:23.682 T:140701446084352  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 77 media: tvshow
2020-05-22 00:15:23.683 T:140701437691648  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:23.684 T:140701437691648  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 3937 media: episode
2020-05-22 00:15:23.686 T:140703153760000  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:23.688 T:140703153760000  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 3938 media: episode
2020-05-22 00:15:23.773 T:140701437691648   ERROR: /opt/Kodi18/.kodi/addons/plugin.video.emby/libraries/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
                                              InsecureRequestWarning)
2020-05-22 00:15:23.952 T:140701437691648   ERROR: Previous line repeats 2 times.
2020-05-22 00:15:23.952 T:140701437691648  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:23.953 T:140701437691648  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 3939 media: episode
2020-05-22 00:15:23.966 T:140701437691648   ERROR: /opt/Kodi18/.kodi/addons/plugin.video.emby/libraries/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
                                              InsecureRequestWarning)
2020-05-22 00:15:24.045 T:140701437691648  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:24.046 T:140701437691648  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 3940 media: episode
2020-05-22 00:15:24.057 T:140703153760000  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:24.058 T:140703153760000  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 3941 media: episode
2020-05-22 00:15:24.065 T:140701437691648   ERROR: /opt/Kodi18/.kodi/addons/plugin.video.emby/libraries/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
                                              InsecureRequestWarning)
2020-05-22 00:15:24.167 T:140703153760000   ERROR: Previous line repeats 1 times.
2020-05-22 00:15:24.168 T:140703153760000  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:24.169 T:140703153760000  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 3943 media: episode
2020-05-22 00:15:24.184 T:140703153760000   ERROR: /opt/Kodi18/.kodi/addons/plugin.video.emby/libraries/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
                                              InsecureRequestWarning)
2020-05-22 00:15:24.295 T:140701437691648  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:24.296 T:140701437691648  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 3944 media: episode
2020-05-22 00:15:24.322 T:140701437691648   ERROR: /opt/Kodi18/.kodi/addons/plugin.video.emby/libraries/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
                                              InsecureRequestWarning)
2020-05-22 00:15:24.383 T:140701437691648  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/VideoLibrary.OnUpdate ]
2020-05-22 00:15:24.383 T:140701437691648  NOTICE: EMBY.objects.monitor ->  [ update/1 ] kodi_id: 3945 media: episode
2020-05-22 00:15:24.397 T:140701437691648   ERROR: /opt/Kodi18/.kodi/addons/plugin.video.emby/libraries/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
                                              InsecureRequestWarning)
2020-05-22 00:15:25.080 T:140701694150400  NOTICE: EMBY.entrypoint.service -> [ onNotification/plugin.video.emby/UserDataChanged ]
2020-05-22 00:15:25.082 T:140701694150400  NOTICE: EMBY.entrypoint.service -> [ UserDataChanged ] {u'UserId': u'cf2b3df2ea9b4cb7ba96a32dd75358fd', u'UserDataList': [{u'ItemId': u'45202', u'IsFavorite': False, u'Played': True, u'LastPlayedDate': u'2020-05-22T06:15:23.0000000+00:00', u'PlaybackPositionTicks': 0, u'PlayCount': 1}, {u'ItemId': u'45201', u'IsFavorite': False, u'Played': True, u'UnplayedItemCount': 0, u'PlaybackPositionTicks': 0, u'PlayCount': 0}, {u'ItemId': u'45214', u'IsFavorite': False, u'Played': True, u'LastPlayedDate': u'2020-05-22T06:15:23.0000000+00:00', u'PlaybackPositionTicks': 0, u'PlayCount': 1}, {u'ItemId': u'45221', u'IsFavorite': False, u'Played': True, u'LastPlayedDate': u'2020-05-22T06:15:23.0000000+00:00', u'PlaybackPositionTicks': 0, u'PlayCount': 1}, {u'ItemId': u'45217', u'IsFavorite': False, u'Played': True, u'LastPlayedDate': u'2020-05-22T06:15:24.0000000+00:00', u'PlaybackPositionTicks': 0, u'PlayCount': 1}, {u'ItemId': u'45228', u'IsFavorite': False, u'Played': True, u'LastPlayedDate': u'2020-05-22T06:15:24.0000000+00:00', u'PlaybackPositionTicks': 0, u'PlayCount': 1}, {u'ItemId': u'45219', u'IsFavorite': False, u'Played': True, u'LastPlayedDate': u'2020-05-22T06:15:24.0000000+00:00', u'PlaybackPositionTicks': 0, u'PlayCount': 1}, {u'ItemId': u'45225', u'IsFavorite': False, u'Played': True, u'LastPlayedDate': u'2020-05-22T06:15:24.0000000+00:00', u'PlaybackPositionTicks': 0, u'PlayCount': 1}, {u'ItemId': u'45226', u'IsFavorite': False, u'Played': True, u'LastPlayedDate': u'2020-05-22T06:15:24.0000000+00:00', u'PlaybackPositionTicks': 0, u'PlayCount': 1}]}
2020-05-22 00:15:25.082 T:140701694150400  NOTICE: EMBY.library -> ---[ userdata:9 ]
2020-05-22 00:15:28.014 T:140701538404096  NOTICE: EMBY.library -> -->[ q:download/140701660326288 ]
2020-05-22 00:15:28.108 T:140701437691648   ERROR: /opt/Kodi18/.kodi/addons/plugin.video.emby/libraries/requests/packages/urllib3/connectionpool.py:791: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
                                              InsecureRequestWarning)
2020-05-22 00:15:29.254 T:140701437691648  NOTICE: EMBY.downloader -> --<[ q:download/140701660326288/9 ]
2020-05-22 00:15:29.255 T:140701437691648  NOTICE: EMBY.downloader -> --[ q:download/remove ] []
2020-05-22 00:15:31.041 T:140701538404096  NOTICE: EMBY.library -> -->[ q:userdata/Episode/140701597853008 ]
2020-05-22 00:15:31.126 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA episode [4023/3937] 45202: Moving the Keys
2020-05-22 00:15:31.360 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA episode [4024/3938] 45214: Handheld Blackjack
2020-05-22 00:15:31.477 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA episode [4027/3941] 45221: The Crash
2020-05-22 00:15:31.588 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA episode [4025/3939] 45217: Prayers and Visions
2020-05-22 00:15:31.720 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA episode [4031/3945] 45228: The Hospital
2020-05-22 00:15:31.835 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA episode [4026/3940] 45219: The Wedding
2020-05-22 00:15:31.937 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA episode [4029/3943] 45225: Alone in This (You Have Me)
2020-05-22 00:15:32.065 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA episode [4030/3944] 45226: That Halloween Night
2020-05-22 00:15:33.066 T:140701437691648  NOTICE: EMBY.database -> [emby] 8 rows updated.
2020-05-22 00:15:33.384 T:140701437691648  NOTICE: EMBY.database -> [video] 8 rows updated.
2020-05-22 00:15:33.432 T:140701437691648  NOTICE: EMBY.library -> --<[ q:userdata/140701597853008 ]
2020-05-22 00:15:34.101 T:140701538404096  NOTICE: EMBY.library -> -->[ q:userdata/Series/140703774673296 ]
2020-05-22 00:15:34.170 T:140701437691648  NOTICE: EMBY.objects.core.tvshows -> USERDATA tvshow [None/77] 45201: Undone
2020-05-22 00:15:35.170 T:140701437691648  NOTICE: EMBY.database -> [emby] 1 rows updated.
2020-05-22 00:15:35.492 T:140701437691648  NOTICE: EMBY.library -> --<[ q:userdata/140703774673296 ]
2020-05-22 00:15:37.171 T:140701538404096  NOTICE: EMBY.library -> --[ sync/2020-05-22T06:13:37z ]
```

**Comment by LukePulverenti** [comment]
I know you said not the kodi one, but i just want to confirm, you don't have the Kodi trakt addon installed, right?

**Comment by blackwind** [comment]
Correct.

**Comment by quickmic** [comment]
I was able to reproduce the issue and will be fixed in next release.

**Comment by quickmic** [comment]
Please check if issue is solved in 4.1.22

**Comment by quickmic** [comment]
fixed
